src/Illuminate/Cache/RedisTaggedCache.php

```diff
@@ -143,8 +143,9 @@ protected function deleteKeysByReference($reference)
      */
     protected function deleteValues($referenceKey)
     {
-        $values = array_unique($this->store->connection()->lrange($referenceKey, 0, -1));
+        $values = array_unique($this->store->connection()->smembers($referenceKey));
 
+        // Remove each tagged cache item
```

Comment:
@GrahamCampbell Yes I have done, perhaps a bit more friendly with your messages. I am only helping to contribute!